### ISSUE TITLE
Storybook 8 compatibility

### ISSUE THREAD

**Issue by joevaugh4n** [linearlabs-workspace/storybook-addon-mock#208]
Hey @nutboltu! I’m one of Storybook’s core maintainers.

Storybook 8 (our next major release) is coming soon, and it introduces some breaking changes that your addon might need to account for.

We’ve listed all the key changes in [a new Storybook 8 migration guide](https://storybook.js.org/docs/8.0/addons/addon-migration-guide#key-changes-for-addons), and we’d encourage you to check it out and see if your addon is impacted.

We also have an addons channel in our Discord community, which we’d like to use to better communicate with addon authors in the future. If you’re not part of the server, consider joining [here](https://discord.gg/wMff8FkF) and message me (@joevaugh4n) so that I can give you the addon creator role! If you’re already in the server, please just ping me and I’ll respond to you there.

Thanks for being a part of our addon community!

Joe

**Comment by lekarev90** [comment]
Yeah. I just have an update on Storybook 8 and have the same problem. Found it with npx storybook@latest doctor

**Comment by MikaKarjunen** [comment]
We are using this addon as well and we have been quite happy with it so far. It would be awesome if you found some time to make it Storybook 8 compatible.

**Comment by nutboltu** [comment]
@lekarev90 @MikaKarjunen I'll prioritise the migration work to support Storybook ^8. I don't have any release date yet, but I will keep you updated here.

**Comment by tonygustafsson** [comment]
How it's released and I'm getting this error when I'm building Storybook 8 with React and Vite:

```
=> Failed to build the preview
RollupError: Expected ',', got ':'
    at getRollupEror (file://./node_modules/vite/node_modules/rollup/dist/es/shared/parseAst.js:375:41)
    at ParseError.initialise (file://./node_modules/vite/node_modules/rollup/dist/es/shared/node-entry.js:11155:28)
    at convertNode (file://./node_modules/vite/node_modules/rollup/dist/es/shared/node-entry.js:12895:10)
    at convertProgram (file://./node_modules/vite/node_modules/rollup/dist/es/shared/node-entry.js:12215:12)
    at Module.setSource (file://./node_modules/vite/node_modules/rollup/dist/es/shared/node-entry.js:14039:24)
    at async ModuleLoader.addModuleSource (file://./node_modules/vite/node_modules/rollup/dist/es/shared/node-entry.js:18697:13)
```

Not if I remove storybook-addon-mock though.

**Comment by janaagaard75** [comment]
This bug in storybook-addon-pseudo-states has the same error message and it started happening with the upgrade to Storybook 8. Linking to it since it might help fix this issue.

https://github.com/chromaui/storybook-addon-pseudo-states/issues/108#issuecomment-1999477683

**Comment by elirov** [comment]
I just contributed to the bounty on this issue:

https://until.dev/bounty/linearlabs-workspace/storybook-addon-mock/208

The current bounty for completing it is $50.00 if it is closed within 1 month, and decreases after that.

**Comment by elirov** [comment]
Any updates on this issue? We're stuck on storybook 7..

**Comment by janaagaard75** [comment]
To anyone else ending up in here, frustrated to be stuck on Storybook 7:

We migrated to [msw-storybook-addon](https://github.com/mswjs/msw-storybook-addon). The setup in a little more complex, but feature wise, it was a drop in replacement for us.

(Note that you have to install v1 of [msw](https://github.com/mswjs/msw) because the add-on it not yet compatible with new version 2. They seem to be working adding support, though.)

**Comment by Jontii** [comment]
Hello ! I've fixed it to the best of my ability and a little sunday afternon. And migrated to storybook v8. I'll create a PR and all of that

**Comment by Jontii** [comment]
Here is a PR for the migration to v8 @elirov 

https://github.com/linearlabs-workspace/storybook-addon-mock/pull/209

**Comment by janaagaard75** [comment]
Really great to see a pull request that could add support for Storybook 8. Does anybody know if @nutboltu will have time to look at the PR? There is a badge with the text "I may be slow to respond" on his profile picture - I don't know about this feature on GitHub, so I don't if this is something added by a bot, or something you set up manually. Does anybody else have merge rights?

**Comment by nutboltu** [comment]
Apologies to everyone for not being active, and a huge shout out to @Jontii for his contribution to v8 compatibility. 
I have published the new version (5.0.0)[https://www.npmjs.com/package/storybook-addon-mock]

**Comment by Jontii** [comment]
This issue should be closed now right ? @nutboltu
